Source/rclUE/Public/Srvs/ROS2SpawnEntitiesSrv.h

```diff
@@ -0,0 +1,271 @@
+                                      in_ros_data.state.data[i].pose.orientation.z,
+                                      in_ros_data.state.data[i].pose.orientation.w));
+        }
+
+        for (int i = 0; i < in_ros_data.state.size; i++)
+        {
+            TwistLinearList.Add(FVector(in_ros_data.state.data[i].twist.linear.x,
+                                        in_ros_data.state.data[i].twist.linear.y,
+                                        in_ros_data.state.data[i].twist.linear.z));
+            TwistAngularList.Add(FVector(in_ros_data.state.data[i].twist.angular.x,
+                                         in_ros_data.state.data[i].twist.angular.y,
+                                         in_ros_data.state.data[i].twist.angular.z));
+        }
+
+        for (int i = 0; i < in_ros_data.state.size; i++)
+        {
+            ReferenceFrameList.Add("");
+            ReferenceFrameList[i].AppendChars(in_ros_data.state.data[i].reference_frame.data,
+                                              in_ros_data.state.data[i].reference_frame.size);
+        }
+
+        for (int i = 0; i < in_ros_data.tags.size; i++)
+        {
+            TagsList.Add("");
+            TagsList[i].AppendChars(in_ros_data.tags.data[i].data, in_ros_data.tags.data[i].size);
+        }
+    }
+
+    void SetROS2(ue_msgs__srv__SpawnEntities_Request& out_ros_data) const
+    {
+        if (out_ros_data.type.data != nullptr)
+        {
+            free(out_ros_data.type.data);
+        }
+        out_ros_data.type.data =
+            (decltype(out_ros_data.type.data))malloc((TypeList.Num()) * sizeof(decltype(*out_ros_data.type.data)));
+        for (int i = 0; i < TypeList.Num(); i++)
+        {
+            {
+                FTCHARToUTF8 strUtf8(*TypeList[i]);
+                int32 strLength = strUtf8.Length();
+                if (out_ros_data.type.data[i].data != nullptr)
+                {
+                    free(out_ros_data.type.data[i].data);
+                }
+                out_ros_data.type.data[i].data = (char*)malloc((strLength + 1) * sizeof(char));
+                memcpy(out_ros_data.type.data[i].data, TCHAR_TO_UTF8(*TypeList[i]), (strLength + 1) * sizeof(char));
+                out_ros_data.type.data[i].size = strLength;
+                out_ros_data.type.data[i].capacity = strLength + 1;
+            }
+        }
+
+        if (out_ros_data.state.data != nullptr)
+        {
+            free(out_ros_data.state.data);
+        }
+        out_ros_data.state.data = (decltype(out_ros_data.state.data))malloc(
+            NameList.Num() * (sizeof(TypeList) + sizeof(NameList) + sizeof(PositionList) + sizeof(OrientationList) +
+                              sizeof(TwistLinearList) + sizeof(TwistAngularList) + sizeof(ReferenceFrameList)));
+        out_ros_data.state.size = NameList.Num();
+        out_ros_data.state.capacity = NameList.Num();
+        for (int i = 0; i < NameList.Num(); i++)
```

Comment:
```suggestion
        for (auto i = 0; i < NameList.Num(); ++i)
```
as configured [here](https://github.com/rapyuta-robotics/UE_tools/blob/devel/CodeGen/CodeGeneratorFromPath.py#L249)?

Reply:
out of curiosity : `++i` is still better than `i++` ? By now, I would have thought that compilers are able to optimize this part automatically

Reply:
Yeah, at some point ago I switched to using prefix increment op in loop due to possible memory cost of the operand, in this case, no practice difference would it bring due to probable optimizing by the compiler for primitive types as you said.
http://www.idinews.com/peeves/prefixIncr.html
https://www.cs.technion.ac.il/users/yechiel/c++-faq/increment-pre-post-speed.html
https://stackoverflow.com/questions/1812990/incrementing-in-c-when-to-use-x-or-x
https://www.tutorialspoint.com/is-there-a-performance-difference-between-iplusplus-and-plusplusi-in-cplusplus-program
https://betterprogramming.pub/stop-using-i-in-your-loops-1f906520d548